libs/ui-watt/src/lib/components/dropdown/watt-dropdown.component.ts

```diff
@@ -0,0 +1,324 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Component,
+  Host,
+  Input,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  AsyncValidatorFn,
+  ControlValueAccessor,
+  FormControl,
+  NgControl,
+  ValidationErrors,
+  ValidatorFn,
+} from '@angular/forms';
+import { MatSelect } from '@angular/material/select';
+import {
+  of,
+  ReplaySubject,
+  Subject,
+  distinctUntilChanged,
+  map,
+  takeUntil,
+} from 'rxjs';
+
+import { WattDropdownOptions } from './watt-dropdown-option';
+import { WattDropdownValue } from './watt-dropdown-value';
+
+@Component({
+  selector: 'watt-dropdown',
+  templateUrl: './watt-dropdown.component.html',
+  styleUrls: ['./watt-dropdown.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class WattDropdownComponent
+  implements ControlValueAccessor, OnInit, OnDestroy
+{
+  /** Subject that emits when the component has been destroyed. */
```

Comment:
Suggestion: Let's figure out an convention how we want our comments formatted, so they can be more consistent.

Reply:
Good point. I'll then suggest `//` for comments that explain a property, code chunk, etc. and TypeDoc kind of comments for API artifacts (property on a component, interface, type, etc.). What do you think?

Will also delete the comment above. We use this pattern in many places and we should be familiar with it by now.